nifi-commons/nifi-security-utils/src/main/java/org/apache/nifi/security/pgp/StandardPGPOperator.java

```diff
@@ -0,0 +1,852 @@
+                new AllowableValue("9", "SHA 384"),
+                new AllowableValue("10", "SHA 512"),
+        };
+    }
+
+    // Values match integer values in org.bouncycastle.bcpg.SymmetricKeyAlgorithmTags
+    static AllowableValue[] getCipherAllowableValues() {
+        return new AllowableValue[]{
+                // 0 - NULL not supported
+                new AllowableValue("1", "IDEA"),
+                new AllowableValue("2", "TRIPLE DES"),
+                new AllowableValue("3", "CAST5"),
+                new AllowableValue("4", "BLOWFISH"),
+                new AllowableValue("6", "DES"),
+                // 6 - SAFER not supported
+                new AllowableValue("7", "AES 128"),
+                new AllowableValue("8", "AES 192"),
+                new AllowableValue("9", "AES 256"),
+                new AllowableValue("10", "TWOFISH"),
+                new AllowableValue("11", "CAMELLIA 128"),
+                new AllowableValue("12", "CAMELLIA 192"),
+                new AllowableValue("13", "CAMELLIA 256")};
+    }
+
+    public static String getCipherDefaultValue() {
+        return String.valueOf(PGPEncryptedData.AES_128);
+    }
+
+    public char[] getPBEPassphrase(PropertyContext context) {
+        if (context.getProperty(PBE_PASSPHRASE).isSet()) {
+            return context.getProperty(PBE_PASSPHRASE).getValue().toCharArray();
+        }
+        return null;
+    }
+
+    public InputStream getSignature(PropertyContext context, FlowFile flowFile) throws PGPException {
+        if (!context.getProperty(SIGNATURE_ATTRIBUTE).isSet()) {
+            throw new PGPException("Signature attribute not set.");
+        }
+
+        final String signature = flowFile.getAttribute(context.getProperty(SIGNATURE_ATTRIBUTE).getValue());
+        try {
+            return new ByteArrayInputStream(Hex.decodeHex(signature));
+        } catch (final DecoderException e) {
+            throw new PGPException("Unable to decode signature.", e);
+        }
+    }
+
+
+    /**
+     * Creates map of public keys from the input stream.
+     *
+     * @param input public key or key ring stream
+     * @param hash value to use as cache key
+     * @return public keys
+     */
+    PGPPublicKeys readPublicKeys(InputStream input, Integer hash) {
+         if (hash != null && publicKeyCache.containsKey(hash)) {
+            return publicKeyCache.get(hash);
+        }
+
+        final PGPPublicKeys keys = new PGPPublicKeys();
+        JcaPGPPublicKeyRingCollection rings;
+
+        try {
+            InputStream decoderStream = PGPUtil.getDecoderStream(input);
+            rings = new JcaPGPPublicKeyRingCollection(decoderStream);
+            input.close();
+        } catch (final IOException | PGPException ignored) {
+            return null;
+        }
+
+        for (final PGPPublicKeyRing ring : rings) {
+            for (final PGPPublicKey key : ring) {
+                for (final Iterator<String> it = key.getUserIDs(); it.hasNext(); ) {
```

Comment:
When using a ECDSA/ECDH pgp key, the public key is created with ECDSA as primary, and ECDH as a sub key. This method fails to detect the ECDH sub key as it does not contain a user id; which means that the keys hash only contains the signing key, not the encryption key. This then fails at ln 742. 
Steps to repeat 
gpg --full-gen-key --expert
Please select what kind of key you want:
   (1) RSA and RSA (default)
   (2) DSA and Elgamal
   (3) DSA (sign only)
   (4) RSA (sign only)
   (7) DSA (set your own capabilities)
   (8) RSA (set your own capabilities)
   (9) ECC and ECC
  (10) ECC (sign only)
  (11) ECC (set your own capabilities)
  (13) Existing key
  (14) Existing key from card
**Your selection? 9**
Please select which elliptic curve you want:
   (1) Curve 25519
   (3) NIST P-256
   (4) NIST P-384
   (5) NIST P-521
   (9) secp256k1
**Your selection? 4**
Please specify how long the key should be valid.
         0 = key does not expire
      <n>  = key expires in n days
      <n>w = key expires in n weeks
      <n>m = key expires in n months
      <n>y = key expires in n years
Key is valid for? (0) 0
Key does not expire at all
Is this correct? (y/N) y

gpg --export --armor --output public-key.asc <key id>